### PR TITLE
feat(luasnip_fn): expand snippet directly;

### PR DESCRIPTION
**Pull request by hinell**
* Expand snippets directly, do not put trigger strings into the buffer
* Setup function now respects userconfig in telescope setup
* Meld stylua config from upstream

### CODE REVIEW

**Comment by benfowler** [comment]
@hinell Thanks for the PR. 

I merged another small PR (performance tweak for Windows users -- thanks @paopaol !), and it's triggered a merge conflict with your PR.

I haven't explicitly done so, but I have a Stylua configuration file I need to commit, which I think is aligned with Neovim Lua conventions. Once I get that done, we could reformat your PR to make the diff a little easier to read. 

How does that sound?

**Comment by benfowler** [comment]
@hinell I've opened benfowler/telescope-luasnip.nvim#20 if that helps you at all.

Again, thanks for bearing with me. We'll try and get this merged ASAP.

**Comment by hinell** [comment]
@benfowler Hi. The #7 PR is a bit wrong.

There is actually a dedicated config for previews that can be reused:, it's called `config.preview.check_mime_type` see 
`h telescope.defaults.preview`

My PR is going to honor this tweak.

File: lua/telescope/_extensions/luasnip.lua [hunk 1]
```diff
@@ -1,154 +1,185 @@
...
+					or default_search_text
+				return {
+					value = entry,
+					display = make_display,
+					ordinal = search_fn(entry),
+					preview_command = function(_, bufnr)
+						local snippet = get_docstring(luasnip, entry.ft, entry.context)
+						if opts.preview.check_mime_type then
+							vim.api.nvim_buf_set_option(bufnr, "filetype", entry.ft)
+						end
```
Comment (hinell) [inline lua/telescope/_extensions/luasnip.lua]: Here you go!

**Comment by hinell** [comment]
@benfowler I've merged `.stylua.tom` from upstrea and formatted it except few places. Checkout.

File: lua/telescope/_extensions/luasnip.lua [hunk 2]
```diff
@@ -1,154 +1,185 @@
...
-                else
-                    print("Snippet '" .. selection.value.context.name .. "'" .. "was selected, but LuaSnip.expandable() returned false")
-                end
-                vim.cmd('stopinsert')
-            end)
-            return true
-        end,
-    }):find()
+  end
+
```
Comment (benfowler) [inline lua/telescope/_extensions/luasnip.lua]: This looks like it's formatted with tabs.

File: lua/telescope/_extensions/luasnip.lua (hunk 1 above)
Comment (benfowler) [reply lua/telescope/_extensions/luasnip.lua]: Cheers. I'm seeing some stuff formatted with tabs though. Not sure if that's intentional?

**Comment by hinell** [comment]
I've expanded tabs to spaces. Checkout.

**Comment by benfowler** [comment]
Hey @hinell , I took the liberty of resolving those merge conflicts for you. If you'd like to quickly cast your eyes over it again, I can merge when you're ready. Thanks again for taking the time to get this change in.

**Comment by hinell** [comment]
@benfowler LGTM. Feel free to merge.

There is a small typo in `error(..)` message btw. Can you fix it after a merge?

**Comment by benfowler** [comment]
Thanks again @hinell for taking the trouble to do this. I've merged, and also fixed that typo. Hope that help. Please shout if you notice anything else wrong.

**Pull request merged**